core/src/main/java/hudson/model/Computer.java

```diff
@@ -958,6 +958,19 @@ public final int countExecutors() {
     }
 
     /**
+     * Gets the read-only snapshot view of all {@link Executor} instances including {@linkplain OneOffExecutor}s.
+     *
+     * @return the read-only snapshot view of all {@link Executor} instances including {@linkplain OneOffExecutor}s.
+     * @since TODO
+     */
+    public List<Executor> getAllExecutors() {
+        List<Executor> result = new ArrayList<>(executors.size() + oneOffExecutors.size());
+        result.addAll(executors);
+        result.addAll(oneOffExecutors);
+        return result;
```

Comment:
Why do you want to add another layer of indirection, plus we have given you a copy so you can do what you want with it